lib/music_theory/note.rb

```diff
@@ -37,10 +38,41 @@ def samples
         phase += sine_wave_cycle
         sample
       end
+      samples = envelope!(samples) if envelope
       samples = distort!(samples) if distort
       samples
     end
 
+    def envelope!(samples)
+      # some function that filters amplitude over the duration of a given note - (to be expanded on)
+      # attack should go from 0 amplitude to max over the course of the first 1/4 of note duration
+      # sustain for the second quarter of the note duration
+      # decay should take the rest of the duration (3/4 and 4/4) of the note to return to 0
+      max_amplitude = samples.map {|s| s.abs }.max
+      slice_size = samples.length / 4
+      attack_slice = samples[0..slice_size]
+      sustain_slice = samples[slice_size..(slice_size * 2)]
+      decay_slice = samples[(slice_size * 2)..(slice_size * 4)]
+
+      amplitude = 0
+      attack_incrementor = max_amplitude / slice_size
+      attack = attack_slice.each do |sample|
+        sample = amplitude
+        amplitude += attack_incrementor
+      end
```

Comment:
This loop isn't doing what you think it is, for a couple reasons. The return of `#each` is always going to be the array that it was called on, and reassigning the `sample` param doesn't change actual value of the sample in the array.